module/VuFind/src/VuFind/View/Helper/Root/JsTranslations.php

```diff
@@ -88,25 +88,35 @@ public function addStrings($new)
     }
 
     /**
-     * Generate JSON from the internal strings.
+     * Generate JSON from the internal strings
      *
      * @return string
      */
     public function getJSON()
     {
-        $parts = [];
-        foreach ($this->strings as $k => $v) {
-            $translation = is_array($v)
-                ? call_user_func_array([$this->transEsc, '__invoke'], $v)
-                : $this->transEsc->__invoke($v);
-            // Special case: do not escape _html translations:
-            if (substr($k, -5) === '_html') {
-                $translation = html_entity_decode($translation);
-            }
-            $parts[] = '"' . addslashes($k) . '": "'
-                . addslashes($translation) . '"';
+        return $this->getJSONFromArray($this->strings);
+    }
+
+    /**
+     * Generate JSON from an array
+     *
+     * @param array $strings Strings to translate (key = js key, value = string to
+     * translate)
+     *
+     * @return string
+     */
+    public function getJSONFromArray(array $strings): string
+    {
+        $view = $this->getView();
+        $translate = $view->plugin('translate');
```

Comment:
It seems a little strange to inject transEsc through the constructor but to pull translate out of the view here. I think we should choose one method or the other and be consistent. (I somewhat prefer the injection approach because it's more explicit and easier to test).

Reply:
Agreed. I tried to keep the constructor back-compatible, but I'll change that.

Reply:
Done. Note that the view and translate view helper are still needed in the tests since TransEsc pulls Translate helper from the view.